pilot/pkg/config/kube/crd/conversion.go

```diff
@@ -36,6 +36,14 @@ func ConvertObject(schema model.ProtoSchema, object IstioObject, domain string)
 		return nil, err
 	}
 	meta := object.GetObjectMeta()
+
+	//// FIXME this is a gross hack to hardcode a service's domain name in kubernetes
+	//if domain == "" {
+	//	domain = "svc.cluster.local"
+	//} else {
+	//	domain = "svc." + domain
+	//}
```

Comment:
remove?

Reply:
No, I want this to be there as a reminder. Ideally, we shouldn't be normalizing for each query to model.DestinationRules. They should be done once when Pilot reads all configs and then retained as is.